platform_channels/test/src/add_pet_details_test.dart

```diff
@@ -9,12 +9,12 @@ import 'package:platform_channels/src/add_pet_details.dart';
 
 void main() {
   group('AddPetDetails tests', () {
-    var petList = <Map>[];
+    var petList = <Map?>[];
```

Comment:
I'm curious what the need for optional `Map`s here is. Also, why the untyped `Map`?

Reply:
The mock handler returns dynamic message, so instead converting `_InternalLinkedHashMap<String, dynamic>` to Map<String, String> using `Map<String, String>.form()` constructor, I decided to keep it Map since it was only for testing purpose. But we can surely change that, or even use Map<String, dynamic>.

Reply:
Sure @domesticmouse  we can  remove the optional `Map`s here  and make `Map` to  `var petList = <Map<String, dynamic>>[];`